src/graph.jl

```diff
@@ -0,0 +1,74 @@
+using JSON
+using SparseArrays
+using LightGraphs
+abstract type AbstractGraph end
+
+struct Graph<:AbstractGraph
+    num_nodes::Int
+    num_edges::Int
+
+    # an adjacency matrix where each entry is the edge id that connects
+    # the two nodes.
+    adj_matrix::SparseMatrixCSC{Int, Int}
+
+    # source and destination of the edge. the index of these arrays are the
+    # edge ids.
+    edge_src::Array{Int, 1}
+    edge_dst::Array{Int, 1}
+
+    # an Array of Arrays where the neighbors of node `i` are found in an array
+    # at index `i`
+    node_neighbors::Array{Array{Int64,1},1}
+
+    # the base SimpleGraph, if we need it
+    simple_graph::SimpleGraph
+end
+
+function Graph(raw_graph::Dict{String, Any},
+               nodes_str::AbstractString = "nodes",
+               adjacency_str::AbstractString = "adjacency",
+               edge_id_str::AbstractString = "id")::Graph
+
+    """ Builds the base Graph object. This is the underlying network of our
```

Comment:
Rather than "never _need_ to be changed," implying that it _could_ be changed but we prefer not to do so, we should probably just say that `Graph` is immutable.

Reply:
ok.